03-routing-and-input.Rmd

```diff
@@ -323,4 +324,40 @@ You can print out the names of all of the properties attached to the request by
 
 // TODO: Conflicts between path, query string, body?
 
+### Form Submit
+
+One common way of developing API's is to submit a form with its inputs using a POST request. For example, to send the following information to the server, we may have the following HTML inputs:
+
+```html
+<h2>Basic submit form</h2>
+
+<pre>
+    <form action="http://localhost:8000/submit" method="post">
+      <input type="text" placeholder="Email" name="email" required>
+      <input type="checkbox" name="bike" value="Bike"> I have a bike
+      <input type="text" placeholder="Bike colour" name="color">
+      <button type="submit">Submit</button>
+    </form>
+</pre>
+```
+
+and  a `/submit` end-point as below:
+
+```{r, echo=FALSE, results='asis'}
+include_r("files/apis/03-06-args.R")
+```
+
+The `name` in the inputs correspond to the `names` of the `req$args` list passed to the Plumber endpoint.
+
+To run/test the example above, just run the file in `files/apis/03-06-args.R` of the repo using `plumber` as follows:
+
+```{r, eval=FALSE}
+p = plumber::plumb("files/apis/03-06-args.R")
+p$run(port = 8000)
```

Comment:
I think it is good to keep this point, though. They will need to run on port 8000 if the HTML is going to work. So let's keep the clarification that it presumes that they're going to run on 8000.

Reply:
Agree.